unlock-app/src/pages/log.js

```diff
@@ -102,16 +114,34 @@ export const mapStateToProps = (
     (a, b) => b.blockNumber - a.blockNumber
   )
 
-  transactionFeed.forEach((tx, i) => {
-    transactionFeed[i].href =
-      chainExplorerUrlBuilders.etherScan(tx.lock) || undefined
-    transactionFeed[i].readableName = humanize(tx.type)
+  // Calculated metadata stored separately from the transactionFeed to avoid
+  // mutating the state by assigning to the objects
+  let metadata = {}
+
+  transactionFeed.forEach(tx => {
+    const href = chainExplorerUrlBuilders.etherScan(tx.lock) || undefined
+    const readableName = humanize(tx.type)
+    let lock = tx.lock
+
+    // Key purchases don't have a lock field, but they do encode the lock the
+    // key is for as part of the key field. So here we create the necessary lock
```

Comment:
We should not do that because the key field shgould be considered an arbitrary string... (it will soon be replaced by the actual TOKEN_ID from the smart contract)

Reply:
You an add the `lock` to the transaction object if needed ;)

Reply:
Right, can we get that in the transaction retriever in web3Service somehow?